src/main/java/org/graylog/integrations/notifications/types/microsoftteams/TeamsEventNotificationConfig.java

```diff
@@ -0,0 +1,154 @@
+    static final String TEAMS_CUSTOM_MESSAGE = "custom_message";
+    static final String TEAMS_ICON_URL = "icon_url";
+    static final String TEAMS_BACKLOG_SIZE = "backlog_size";
+    static final String TEAMS_COLOR = "color";
+
+    @JsonProperty(TEAMS_BACKLOG_SIZE)
+    public abstract long backlogSize();
+
+    @JsonProperty(TEAMS_COLOR)
+    @NotBlank
+    public abstract String color();
+
+    @JsonProperty(FIELD_WEBHOOK_URL)
+    @NotBlank
+    public abstract String webhookUrl();
+
+    @JsonProperty(TEAMS_CUSTOM_MESSAGE)
+    public abstract String customMessage();
+
+    @JsonProperty(TEAMS_ICON_URL)
+    @Nullable
+    public abstract String iconUrl();
+
+    @Override
+    @JsonIgnore
+    public JobTriggerData toJobTriggerData(EventDto dto) {
+        return EventNotificationExecutionJob.Data.builder().eventDto(dto).build();
+    }
+
+    public static Builder builder() {
+        return Builder.create();
+    }
+
+    @Override
+    @JsonIgnore
+    public ValidationResult validate() {
+        ValidationResult validation = new ValidationResult();
+
+        URI webhookUri;
+        try {
+            webhookUri = new URI(webhookUrl());
+            if (webhookUri.getHost().toLowerCase().contains("office")) {
+                if (!TEAMS_PATTERN.matcher(webhookUrl()).find()) {
+                    validation.addError(FIELD_WEBHOOK_URL, INVALID_TEAMS_URL_ERROR_MESSAGE);
+                }
+            }
+
+        } catch (Exception ex) {
+            validation.addError(FIELD_WEBHOOK_URL, INVALID_WEBHOOK_ERROR_MESSAGE);
+        }
+
+        if (backlogSize() < 0) {
+            validation.addError(TEAMS_BACKLOG_SIZE, INVALID_BACKLOG_ERROR_MESSAGE);
+        }
+
+        return validation;
+    }
+
+    @AutoValue.Builder
+    public static abstract class Builder implements EventNotificationConfig.Builder<Builder> {
+        @JsonCreator
+        public static Builder create() {
+
+            return new AutoValue_TeamsEventNotificationConfig.Builder()
+                    .type(TYPE_NAME)
+                    .color(DEFAULT_HEX_COLOR)
+                    .webhookUrl(WEB_HOOK_URL)
+                    .customMessage(DEFAULT_CUSTOM_MESSAGE)
+                    .backlogSize(DEFAULT_BACKLOG_SIZE);
+        }
+
+        @JsonProperty(TEAMS_COLOR)
+        public abstract Builder color(String color);
+
+        @JsonProperty(FIELD_WEBHOOK_URL)
+        public abstract Builder webhookUrl(String webhookUrl);
+
+        @JsonProperty(TEAMS_CUSTOM_MESSAGE)
+        public abstract Builder customMessage(String customMessage);
+
+        @JsonProperty(TEAMS_ICON_URL)
+        public abstract Builder iconUrl(String iconUrl);
+
+        @JsonProperty(TEAMS_BACKLOG_SIZE)
+        public abstract Builder backlogSize(long backlogSize);
+
+        public abstract TeamsEventNotificationConfig build();
+    }
+
+    @Override
+    public EventNotificationConfigEntity toContentPackEntity(EntityDescriptorIds entityDescriptorIds) {
```

Comment:
Has anyone been able to export/import instances of the notification as a content pack yet? I can create the content pack, but when I attempt to install it, this error occurs. I've not found the cause yet.
```
2022-06-13 15:20:55,970 ERROR: org.graylog2.shared.rest.exceptionmappers.AnyExceptionClassMapper - Unhandled exception in REST resource
org.graylog2.contentpacks.exceptions.ContentPackException: Failed to install content pack <3b8647d2-f5bd-4556-a479-21fa3c053659/1>
	at org.graylog2.contentpacks.ContentPackService.installContentPack(ContentPackService.java:159) ~[classes/:?]
	at org.graylog2.contentpacks.ContentPackService.installContentPack(ContentPackService.java:100) ~[classes/:?]
	at org.graylog2.rest.resources.system.contentpacks.ContentPackResource.installContentPack(ContentPackResource.java:293) ~[classes/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:469) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:391) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:80) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255) [jersey-server-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234) [jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680) [jersey-server-2.32.jar:?]
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356) [jersey-container-grizzly2-http-2.32.jar:?]
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:200) [grizzly-http-server-2.4.4.jar:2.4.4]
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [metrics-core-4.1.9.jar:4.1.9]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.IllegalArgumentException: Could not resolve type id 'teams-notification-v1' as a subtype of [simple type, class org.graylog.events.contentpack.entities.EventNotificationConfigEntity]: known type ids = [email-notification-v1, http-notification-v1, legacy-alarm-callback-notification-v1] (for POJO property 'config')
 at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: org.graylog.events.contentpack.entities.AutoValue_NotificationEntity$Builder["config"])
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3751) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:3669) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at org.graylog.events.contentpack.facade.NotificationFacade.decode(NotificationFacade.java:99) ~[classes/:?]
	at org.graylog.events.contentpack.facade.NotificationFacade.createNativeEntity(NotificationFacade.java:90) ~[classes/:?]
	at org.graylog2.contentpacks.ContentPackService.installContentPack(ContentPackService.java:150) ~[classes/:?]
	... 29 more
Caused by: com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Could not resolve type id 'teams-notification-v1' as a subtype of [simple type, class org.graylog.events.contentpack.entities.EventNotificationConfigEntity]: known type ids = [email-notification-v1, http-notification-v1, legacy-alarm-callback-notification-v1] (for POJO property 'config')
 at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: org.graylog.events.contentpack.entities.AutoValue_NotificationEntity$Builder["config"])
	at com.fasterxml.jackson.databind.exc.InvalidTypeIdException.from(InvalidTypeIdException.java:43) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.DeserializationContext.invalidTypeIdException(DeserializationContext.java:1635) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownTypeId(DeserializationContext.java:1187) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.jsontype.impl.TypeDeserializerBase._handleUnknownTypeId(TypeDeserializerBase.java:291) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.jsontype.impl.TypeDeserializerBase._findDeserializer(TypeDeserializerBase.java:162) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedForId(AsPropertyTypeDeserializer.java:113) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:97) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.deser.AbstractDeserializer.deserializeWithType(AbstractDeserializer.java:254) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeSetAndReturn(MethodProperty.java:167) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.vanillaDeserialize(BuilderBasedDeserializer.java:269) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserialize(BuilderBasedDeserializer.java:193) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3746) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:3669) ~[jackson-databind-2.9.10.4.jar:2.9.10.4]
	at org.graylog.events.contentpack.facade.NotificationFacade.decode(NotificationFacade.java:99) ~[classes/:?]
	at org.graylog.events.contentpack.facade.NotificationFacade.createNativeEntity(NotificationFacade.java:90) ~[classes/:?]
	at org.graylog2.contentpacks.ContentPackService.installContentPack(ContentPackService.java:150) ~[classes/:?]
	... 29 more
```

Reply:
Also, the `entityDescriptorIds` appear to not be used here. Is there any reason for that? What are they for? I have not had a chance to research it yet.

Reply:
Content pack install error fixed in https://github.com/Graylog2/graylog-plugin-integrations/pull/1064/commits/5daee06cf12eb93b0f745c9dc8e981a244a9f44e

The `entityDescriptorIds` are also not used for other notifications, so that is probably fine here.